src/test/java/org/assertj/core/internal/objects/Objects_assertIsEqualToComparingFieldByFieldRecursive_Test.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2015 the original author or authors.
+ */
+package org.assertj.core.internal.objects;
+
+import static org.assertj.core.error.ShouldBeEqual.shouldBeEqual;
+import static org.assertj.core.test.TestData.someInfo;
+import static org.assertj.core.test.TestFailures.failBecauseExpectedAssertionErrorWasNotThrown;
+import static org.mockito.Mockito.verify;
+
+import org.assertj.core.api.AssertionInfo;
+import org.assertj.core.internal.ObjectsBaseTest;
+import org.assertj.core.presentation.StandardRepresentation;
+import org.junit.Test;
+
+public class Objects_assertIsEqualToComparingFieldByFieldRecursive_Test extends ObjectsBaseTest {
+
+  @Test
+  public void should_be_able_to_compare_objects_recursivly() {
+    Person actual = new Person();
+    actual.name = "John";
+    actual.home.address.number = 1;
+    
+    Person other = new Person();
+    other.name = "John";
+    other.home.address.number = 1;
+    
+    objects.assertIsEqualToComparingFieldByFieldRecursively(someInfo(), actual, other); 
+  }
+  
```

Comment:
would be nice to have a test with objects cycle - adding a `neighbour` field to `Person` should be enough.